webui/react/src/pages/ExperimentList.tsx

```diff
@@ -81,13 +81,119 @@ const ExperimentList: React.FC = () => {
     },
   );
   const initSorter = storage.getWithDefault(STORAGE_SORTER_KEY, { ...defaultSorter });
-  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
-  const [ total, setTotal ] = useState(0);
+  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ filters, setFilters ] = useState<ExperimentFilters>(initFilters);
-  const [ sorter, setSorter ] = useState(initSorter);
+  const [ isUrlParsed, setIsUrlParsed ] = useState<boolean>(false);
+  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
   const [ search, setSearch ] = useState('');
-  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ selectedRowKeys, setSelectedRowKeys ] = useState<string[]>([]);
+  const [ sorter, setSorter ] = useState(initSorter);
+  const [ total, setTotal ] = useState(0);
+
+  /*
+   * When filters changes update the page URL.
+   */
+  useEffect(() => {
+    if (!isUrlParsed) return;
+
+    const searchParams: Record<string, string> = {
+      archived: filters.showArchived ? '1' : '0',
+      limit: pagination.limit.toString(),
+      offset: pagination.offset.toString(),
+      sortDesc: sorter.descend ? '1' : '0',
+    };
+    if (filters.labels) {
+      searchParams.labels = filters.labels.join(',');
+    }
+    if (search) {
+      searchParams.search = search;
+    }
+    if (typeof sorter.key === 'string') {
+      searchParams.sortKey = sorter.key;
+    }
+    if (filters.states) {
+      searchParams.states = filters.states.join(',');
+    }
+    if (filters.username) {
+      searchParams.username = filters.username;
+    }
+
+    const searchParamsString = new URLSearchParams(searchParams).toString();
```

Comment:
I don't like the fact that `toString` is forcing the conversion of URL special chars (eg: `...&labels=x,y,z&...` becomes  `...&labels=x%2Cy%2Cz&...`.

Not sure if you have any suggestion about this, because the only alternative is to manually "merge" that URL params object instead of relying on `toString`.

Reply:
I think param values are always encoded to avoid injections.

If we manually "merge" or override this behavior, then it would be on us to parse and validate the values and escape and/or ignore certain characters before we feed it to the API.

Another option can be to use the special characters that are left alone. Maybe the period can be an option, where it looks close enough to a comma.
 
This [link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs_encodeuricomponent) shows that the reserved special characters that ARE encoded:
`;,/?:@&=+$#`

where as these are NOT escaped:
`-_.!~*'()`

Reply:
Another thing to consider is that labels can contain all of those special characters... 
<img width="749" alt="Schermata 2021-02-24 alle 09 54 23" src="https://user-images.githubusercontent.com/5413702/108974986-4fb6b000-7686-11eb-8fa8-ae8b63288f12.png">

so having a query string with &labels=x.y will not work.

Reply:
Updated to build/parse url like this `...&label=x&label=y&label=z...`